src/VegaWrapper2.js

```diff
@@ -0,0 +1,371 @@
+                }
+                urlParts.query = {
+                    format: 'json',
+                    formatversion: '2',
+                    action: 'query',
+                    prop: 'revisions',
+                    rvprop: 'content',
+                    titles: urlObj.title
+                };
+
+                urlParts.pathname = '/w/api.php';
+                options.addCorsOrigin = true;
+                break;
+
+            case 'tabular':
+            case 'map':
+                // { type: 'tabular', title: 'Data.tab' [, lang: 'en'] }
+                // { type: 'map', title: 'Data.map' [, lang: 'en'] }
+                // Get content of a wiki page belonging to Data namespace
+                // Uses mediawiki api, and extract the content after the request
+                // Query value must be a valid MediaWiki title string, so we ensure there
+                // is no pipe symbol or \x1F and the title ends with .tab or .map
+                if (!/^[^|\x1F]+\.(tab|map)$/.test(urlObj.title)) {
+                    throw new Error(urlObj.type + ': invalid title');
+                }
+                urlParts.query = {
+                    format: 'json',
+                    formatversion: '2',
+                    action: 'jsondata',
+                    title: urlObj.title
+                };
+                if (urlObj.lang || this.languageCode) {
+                    urlParts.query.uselang = urlObj.lang || this.languageCode;
+                }
+
+                urlParts.pathname = '/w/api.php';
+                options.addCorsOrigin = true;
+                break;
+
+            case 'wikifile':
+                // {type: “wikifile”, title: “Einstein_1921.jpg”, [width=100, height=100]}
+                // Get an image for the graph, e.g. from commons, by using Special:Redirect
+                urlParts.pathname = '/wiki/Special:Redirect/file/' + urlObj.title;
+                urlParts.query = {};
+                if(urlObj.width) urlParts.query.width = urlObj.width;
+                if(urlObj.height) urlParts.query.height = urlObj.height;
+                break;
+
+            case 'wikidatasparql':
+                // {type: “wikidatasparql”, query: "..."}
+                // Runs a SPARQL query, converting it to
+                // https://query.wikidata.org/bigdata/namespace/wdq/sparql?format=json&query=...
+                this._overrideHostAndProtocol(urlParts, urlObj);
+                if (!urlObj.query) {
+                    throw new Error('wikidatasparql: missing query parameter');
+                }
+                urlParts.query = {query: urlObj.query};
+                urlParts.pathname = '/bigdata/namespace/wdq/sparql';
+                options.headers = Object.assign(options.headers || {}, {'Accept': 'application/sparql-results+json'});
+                break;
+
+            case 'geoshape':
+            case 'geoline':
+                // {type: “geoshape”, [ids: “Q16,Q30” | query:’...’] }
```

Comment:
I don't think comma-separated values is a good approach here, lets use proper json, e.g.:
```suggestion
                // {type: “geoshape”, [ids: ["Q16", "Q30"] | query: "..."] }
```